### ISSUE TITLE
[Spike] flipside / midguard data for rFOX dashboard for total fees collected this epoch 

### ISSUE THREAD

**Issue by 0xean** [shapeshift/web#6951]
![Screenshot 2024-05-20 at 6 11 59 PM](https://github.com/shapeshift/web/assets/15096737/09ffc20c-5510-4ddd-94f3-cd805a1a9254)

Determine if we can source total fees in the current epoch using existing APIs. Possible solutions include flipside or midguard.

- rumors of https://affiliates.ninerealms.com/ss/90/days being built on flipside which may mean we can consume that same data to be able to dermine epoch to date rune earnings 
- epoch's are probably 28 days, so assume we need up to 28 days max of history. 
- document findings here and create follow up ticket for implementation into the UI

**Comment by tolltalcrypto** [comment]
https://discord.com/channels/838986635756044328/1185285947043090433/1242251200091521097  tc dev discord shapeshift private channel.

https://affiliates.ninerealms.com/  confirmed coming from midgard /action endpoint.

**Comment by woodenfurniture** [comment]
This could work
https://affiliates.ninerealms.com/ss/28/days?_data=routes%2F%24affiliate.%24count.%24type

**Comment by woodenfurniture** [comment]
pluto or familiarcow will get back to us in the private discord channel with further info

**Comment by woodenfurniture** [comment]
Actions endpoint only works if we want to page through every transaction the org has done for the 28 day epoch, for every client. Might not be scalable so we'd either need an aggregation endpoint of some kind, or build our own.

e.g
https://midgard.ninerealms.com/v2/actions?affiliate=ss,fromTimestamp=1718006345,timestamp=1718066345

Docs here
https://midgard.ninerealms.com/v2/doc#operation/GetActions

**Comment by woodenfurniture** [comment]
Actions are:
1. implement unblocker react query to page through all actions for shapeshift affiliate, to aggregate the total RUNE revenue
2. optimize the above with a custom index on the backend for scalability

I'll ticket.

**Comment by woodenfurniture** [comment]
Short-term unblock ticketed here:
https://github.com/shapeshift/web/issues/7117

Scalable solution ticketed here:
https://github.com/shapeshift/web/issues/7116

**Comment by woodenfurniture** [comment]
Context on flipside dashboard spike:
https://discord.com/channels/554694662431178782/886041582422356018/1251015248606134423

TL;DR is that its very difficult and complex to reconcile streaming swaps and inbound/outbound txs in flipside, making the affiliate revenue calculation hard. There may be a simpler way to report on affiliate revs other than reconciling affiliate txs to the affiliate address, but none was identified in this spike.
